tracer/src/Datadog.Tracer.Native/iast/dataflow.cpp

```diff
@@ -322,6 +323,128 @@ void Dataflow::LoadAspects(WCHAR** aspects, int aspectsLength, UINT32 enabledCat
+{
+    auto securityControlsConfig = shared::GetEnvironmentValue(environment::security_controls_configuration);
+    if (securityControlsConfig.size() > 0)
+    {
+        DataflowAspectClass* aspectClass = nullptr;
+
+        trace::Logger::Debug("Dataflow::LoadSecurityControls -> Processing Security Controls Config... ",
+                             securityControlsConfig);
+        auto securityControls = shared::Split(securityControlsConfig, ';');
+        for (auto securityControlLine : securityControls)
+        {
+            auto securityControl = shared::Trim(securityControlLine);
+            if (securityControl.size() == 0 || securityControl[0] == '#')
+            {
+                continue;
+            }
+
+            auto parts = shared::Split(securityControl, ':');
+            if (parts.size() < 5)
+            {
+                trace::Logger::Warn("Dataflow::LoadSecurityControls -> Detected invalid Security Control: ",
+                                    securityControl);
+                continue;
+            }
+
+            int part = -1;
+            SecurityControlType securityControlType = SecurityControlType::Unknown;
+            if ((int) parts.size() > ++part) // Security control kind
+            {
+                securityControlType = ParseSecurityControlType(parts[part]);
+            }
+            if (securityControlType == SecurityControlType::Unknown)
+            {
+                trace::Logger::Warn("Dataflow::LoadSecurityControls -> Detected invalid Security Control type: ",
+                                    parts[part], " in ",
+                                    securityControl);
+                continue;
+            }
+
+            UINT32 secureMarks = 0;
+            if ((int) parts.size() > ++part) // Vulnerability type
+            {
+                for (auto vulnPart : Split(shared::ToString(parts[part]), ","))
+                {
+                    auto vuln = ParseVulnerabilityType(shared::ToString(vulnPart));
+                    if (vuln == VulnerabilityType::None)
+                    {
+                        trace::Logger::Warn(
+                            "Dataflow::LoadSecurityControls -> Detected invalid Security Control vulnerability type: ",
+                            vulnPart, " in ",
+                            securityControl);
+                        continue;
+                    }
+
+                    secureMarks |= (UINT32) vuln;
+                }
+            }
+            if (secureMarks == 0)
+            {
+                trace::Logger::Warn(
+                    "Dataflow::LoadSecurityControls -> Detected invalid Security Control vulnerability types: ",
+                    securityControl);
+                continue;
+            }
+
+            auto targetAssembly = parts[++part];
+            auto targetType = parts[++part];
+            auto targetMethodPart = parts[++part];
+
+            std::vector<int> parameterIndexes;
```

Comment:
if you can, you can passe the number of elements to `parameterIndexes` to avoid temporary allocations.